e2e/disruptors/pod_e2e_test.go

```diff
@@ -154,16 +157,45 @@ func Test_PodDisruptor(t *testing.T) {
 				go func() {
 					err := tc.injector(disruptor)
 					if err != nil {
-						t.Logf("failed to setup disruptor: %v", err)
+						t.Errorf("failed to setup disruptor: %v", err)
```

Comment:
🤔 I'm not sure I understand what you mean by 
> if the test ends before this sentence is executed
could you elaborate a bit?
I'm happy to change it back if we have a reason to, but I though doing log+return would mark the test as successful and hide the problem if the test is not run with `-v`.

Reply:
Initially, the code was as you have it now. But I got weird errors at some point because this grouting may be still running when the test ended. 

I'm not sure but I think this could reproduce the situation:  Imagine the agent container cannot be created. The default timeout for the container to become ready is `30s`. Therefore the fault injection will fail and the `Errorf` will be called.

If the check fails before this timeout expires, the test will end and the `Errorf` will be called (due to the failure described above) after the test ends and there will be a weird error in the testing library. 

I guess the right solution is for the test to wait for the goroutine to end. We can do this in a followup PR or I suggest leaving the code as it is and adding a TODO for fixing the synchronization.

Reply:
Thanks for the explanation, it makes sense now. I agree to tackle this when we refactor these tests, I'll add a TODO so we don't forget.